packages/graph/src/-private/edges/resource.ts

```diff
@@ -40,16 +40,19 @@ export function createResourceEdge(definition: UpgradedMeta, identifier: StableR
   };
 }
 
-export function legacyGetResourceRelationshipData(source: ResourceEdge): ResourceRelationship {
+export function legacyGetResourceRelationshipData(source: ResourceEdge, getRemoteState: boolean): ResourceRelationship {
   source.accessed = true;
   let data: StableRecordIdentifier | null | undefined;
   const payload: ResourceRelationship = {};
-  if (source.localState) {
+  if (getRemoteState && source.remoteState) {
+    data = source.remoteState;
+  } else if (!getRemoteState && source.localState) {
     data = source.localState;
   }
-  if (source.localState === null && source.state.hasReceivedData) {
+  if (((getRemoteState && source.remoteState === null) || source.localState === null) && source.state.hasReceivedData) {
```

Comment:
@richgt in my testing (on 4.13, backported from 5.3), this line seems incorrect in the situation where a relationship was non-null, but is set to null locally and not yet saved.  I think the line should be:
```
if ((getRemoteState ? source.remoteState : source.localState) === null) && source.state.hasReceivedData) {
```

Reply:
assuming:
- `remoteState !== null`
- `localState === null`

for `getRemoteState === true`

- line 48 gives us the remote value
- check on line 52 becomes: (((true && false) || true) && true) which evals to `true`
- we assign `null` to the data incorrectly

for `getRemoteState === false`

- conditions on lines 47 and 49 are false
- line 52 evals to (((false && false) || true) && true) which evals to `true`
- we assign `null` to the data correctly

I believe the mistake here is actually that the check on line 52 is an `if` instead of an `else if`